subsys/modem/backends/modem_backend_uart.c

```diff
@@ -9,15 +9,12 @@
 
 #include <zephyr/modem/backend/uart.h>
 
-#include <zephyr/logging/log.h>
-LOG_MODULE_REGISTER(modem_backend_uart, CONFIG_MODEM_MODULES_LOG_LEVEL);
-
 #include <string.h>
 
 static void modem_backend_uart_receive_ready_handler(struct k_work *item)
 {
-	struct modem_backend_uart *backend =
-		CONTAINER_OF(item, struct modem_backend_uart, receive_ready_work);
+	struct modem_backend_uart *backend = CONTAINER_OF(
+		(struct k_work_delayable *)item, struct modem_backend_uart, receive_ready_work);
```

Comment:
There is the `k_work_delayable_from_work(struct k_work *work);` that you should use instead of direct cast.